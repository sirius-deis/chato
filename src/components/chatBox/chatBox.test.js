import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import ChatBox from "./chatBox";
import { createStore } from "redux";

const INITIAL_STATE = { user: { user: { id: 1 } } };

const messages = [
  { message: "Hello", senderId: 1, createdAt: "" },
  { message: "Hi", senderId: 2, createdAt: "" },
];

const props = {
  id: 1,
  messages,
  title: "Some chat",
  unreadMessagesCount: 2,
};

const reducer = (state = INITIAL_STATE) => {
  return state;
};

const store = createStore(reducer);

describe("ChatBox component", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <BrowserRouter>
        <Provider store={store}>
          <ChatBox {...props} />
        </Provider>
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
  it("should render messages", () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <ChatBox {...props} />
        </Provider>
      </BrowserRouter>
    );
    expect(screen.getByText(/Hello/)).toBeInTheDocument();
  });
  it("should render a chat title", () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <ChatBox {...props} />
        </Provider>
      </BrowserRouter>
    );
    expect(screen.getByText(/Some chat/)).toBeInTheDocument();
  });
  it("should render a chat icon with a letter as there is no picture provided", () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <ChatBox {...props} />
        </Provider>
      </BrowserRouter>
    );
    expect(screen.getByText("S")).toBeInTheDocument();
  });
  it("should render without online sign", () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <ChatBox {...props} />
        </Provider>
      </BrowserRouter>
    );
    expect(screen.queryByTestId("status")).not.toBeInTheDocument();
  });
  it("should render with online sign", () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <ChatBox {...props} isOnline />
        </Provider>
      </BrowserRouter>
    );
    expect(screen.getByTestId("status")).toBeInTheDocument();
  });
});
