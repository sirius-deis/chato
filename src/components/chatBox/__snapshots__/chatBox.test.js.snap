// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatBox component should match snapshot 1`] = `
<div>
  <div
    class="sc-aXZVg kDvmMR"
  >
    <div
      class="sc-gEvEer cxDxNl"
    >
      <div>
        S
      </div>
    </div>
    <div
      class="sc-fqkvVR dPxtmS"
    >
      <div
        class="sc-imWYAI kbLuvu"
      >
        <h6
          class="sc-iGgWBj jWoWlb"
        >
          Some chat
        </h6>
        <div
          class="sc-gsFSXq cHstVC"
        >
          NaNNaNNaN
        </div>
      </div>
      <div
        class="sc-imWYAI eqgqaX"
      >
        <div
          class="sc-dcJsrY edasem"
        >
          Hello
        </div>
      </div>
    </div>
    <span
      class="sc-kAyceB cXddJt"
    >
      2
    </span>
  </div>
</div>
`;
